### PR TITLE
Friends Routes Tested

### PR DESCRIPTION
**Pull request by terryzfeng**
Working Friends Routes.
Some minor changes here and there but no clue where
Fixed shared hugs.

### CODE REVIEW

**Review by Rick10101221 — COMMENTED** [review]
lgtm

File: backend/model/Hugs.js [hunk 1]
```diff
@@ -261,6 +262,7 @@ const UpdateHugAPI = {
       .collection("notifications");
     requestIdRef = currUserNotifRef.where("hug_id", "==", hugId);
     Notifications.NotificationsAPI.deleteNotification(requestIdRef);
+     */
   },
 
   updateUserHugCount: function (hugId) {
```
Comment (evanjserrano) [inline backend/model/Hugs.js:268]: Should this be async?

File: backend/routes/hugsRoutes.js [hunk 2]
```diff
@@ -34,22 +33,20 @@ async function retrieveUserData(req, res, uid) {
 // Routes
 // TODO: BROKEN. CREATE HUG IS NOT WORKING.
```
Comment (evanjserrano) [inline backend/routes/hugsRoutes.js:34]: Is this still broken?

**Review by evanjserrano — COMMENTED** [review]
Is hugRoutes still broken?

File: backend/model/Hugs.js (hunk 1 above)
Comment (evansuong) [reply backend/model/Hugs.js:268]: this is fine even though it deviates from the other functions bc there is no await statement, instead this is handling async logic with .then()

**Pull request merged**